[v2][Android] there is an error while merging dex archives

### ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
I created my react native project via 'react-native init' and just installed it.    

And I tried to follow the installation guide but I got an error on ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.     
   
So I ran 'gradlew clean' and rebuilt my project, but it wasn't fixed.
   
The error log is like below.
```
λ npm run android                                                                                                                                                            
                                                                                                                                                                             
> app@0.0.1 android C:\Users\ytw0728\Desktop\Farerpath\mobile\app                                                                                                            
> cd ./android && gradlew app:assembleDebug && gradlew installDebug                                                                                                          
                                                                                                                                                                             
                                                                                                                                                                             
> Configure project :react-native-navigation                                                                                                                                 
downloadRobolectricDependencies into C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\build\robolectric-3.5.1-dependencies                                              
                                                                                                                                                                             
> Task :app:transformDexArchiveWithExternalLibsDexMergerForDebug FAILED                                                                                                      
D8: Program type already present: android.support.design.widget.CoordinatorLayout$Behavior                                                                                   
                                                                                                                                                                             
                                                                                                                                                                             
FAILURE: Build failed with an exception.                                                                                                                                     
                                                                                                                                                                             
* What went wrong:                                                                                                                                                           
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.                                                                                       
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\trans
forms\dexBuilder\debug\5.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\6.jar, C:\Users\ytw0728\Desktop\Farer
path\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\7.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\d
exBuilder\debug\8.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\9.jar, C:\Users\ytw0728\Desktop\Farerpath\mo
bile\app\android\app\build\intermediates\transforms\dexBuilder\debug\10.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuil
der\debug\11.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\12.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile
\app\android\app\build\intermediates\transforms\dexBuilder\debug\18.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\
debug\19.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\20.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app
\android\app\build\intermediates\transforms\dexBuilder\debug\21.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debu
g\22.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\23.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\and
roid\app\build\intermediates\transforms\dexBuilder\debug\24.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\25
.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\26.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android
\app\build\intermediates\transforms\dexBuilder\debug\27.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\28.jar
, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\31.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app
\build\intermediates\transforms\dexBuilder\debug\33.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\34.jar, C:
\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\35.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\bui
ld\intermediates\transforms\dexBuilder\debug\36.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\37.jar, C:\Use
rs\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\38.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\i
ntermediates\transforms\dexBuilder\debug\39.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\40.jar, C:\Users\y
tw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\41.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\inter
mediates\transforms\dexBuilder\debug\89.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\90.jar, C:\Users\ytw07
28\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\91.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermedi
ates\transforms\dexBuilder\debug\92.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\93.jar, C:\Users\ytw0728\D
esktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\94.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates
\transforms\dexBuilder\debug\95.jar, C:\Users\ytw0728\Desktop\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\96.jar, C:\Users\ytw0728\Deskt
op\Farerpath\mobile\app\android\app\build\intermediates\transforms\dexBuilder\debug\97.jar                                                                                   
                                                                                                                                                                             
* Try:                                                                                                                                                                       
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.                         
                                                                                                                                                                             
* Get more help at https://help.gradle.org                                                                                                                                   
                                                                                                                                                                             
BUILD FAILED in 28s                                                                                                                                                          
38 actionable tasks: 3 executed, 35 up-to-date                                                                                                                               
npm ERR! code ELIFECYCLE                                                                                                                                                     
npm ERR! errno 1                                                                                                                                                             
npm ERR! app@0.0.1 android: `cd ./android && gradlew app:assembleDebug && gradlew installDebug`                                                                              
npm ERR! Exit status 1                                                                                                                                                       
npm ERR!                                                                                                                                                                     
npm ERR! Failed at the app@0.0.1 android script.                                                                                                                             
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.                                                                           
                                                                                                                                                                             
npm ERR! A complete log of this run can be found in:                                                                                                                         
npm ERR!     C:\Users\ytw0728\AppData\Roaming\npm-cache\_logs\2019-01-01T20_15_44_946Z-debug.log                                                                             
                                                                                                                                                                             
```   
   
   
   
and here is my setting.gradle
```
rootProject.name = 'app'

include ':app'

include ':react-native-navigation' // for react-native-navigation
project(':react-native-navigation').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-navigation/lib/android/app/') // for react-native-navigation
```
   
   
build.gradle    
```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
    ext {
        buildToolsVersion = "27.0.3"
        // minSdkVersion = 16
        minSdkVersion = 19 // for react-native-navigation
        compileSdkVersion = 27
        targetSdkVersion = 26
        supportLibVersion = "27.1.1"
    }
    repositories {
        google()
        mavenLocal() // for react-native-navigation
        mavenCentral() // for react-native-navigation
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.1.4'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        google()
        mavenCentral() // for react-native-navigation
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven { url 'https://jitpack.io' } // for react-native-navigation
    }
}


task wrapper(type: Wrapper) {
    gradleVersion = '4.4'
    distributionUrl = distributionUrl.replace("bin", "all")
}
```
    
   
   
app/build.gradle   
```
apply plugin: "com.android.application"

import com.android.build.OutputFile

/**
 * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
 * and bundleReleaseJsAndAssets).
 * These basically call `react-native bundle` with the correct arguments during the Android build
 * cycle. By default, bundleDebugJsAndAssets is skipped, as in debug/dev mode we prefer to load the
 * bundle directly from the development server. Below you can see all the possible configurations
 * and their defaults. If you decide to add a configuration block, make sure to add it before the
 * `apply from: "../../node_modules/react-native/react.gradle"` line.
 *
 * project.ext.react = [
 *   // the name of the generated asset file containing your JS bundle
 *   bundleAssetName: "index.android.bundle",
 *
 *   // the entry file for bundle generation
 *   entryFile: "index.android.js",
 *
 *   // whether to bundle JS and assets in debug mode
 *   bundleInDebug: false,
 *
 *   // whether to bundle JS and assets in release mode
 *   bundleInRelease: true,
 *
 *   // whether to bundle JS and assets in another build variant (if configured).
 *   // See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Build-Variants
 *   // The configuration property can be in the following formats
 *   //         'bundleIn${productFlavor}${buildType}'
 *   //         'bundleIn${buildType}'
 *   // bundleInFreeDebug: true,
 *   // bundleInPaidRelease: true,
 *   // bundleInBeta: true,
 *
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true,
 *   // The configuration property can be in the following formats
 *   //         'devDisabledIn${productFlavor}${buildType}'
 *   //         'devDisabledIn${buildType}'
 *
 *   // the root of your project, i.e. where "package.json" lives
 *   root: "../../",
 *
 *   // where to put the JS bundle asset in debug mode
 *   jsBundleDirDebug: "$buildDir/intermediates/assets/debug",
 *
 *   // where to put the JS bundle asset in release mode
 *   jsBundleDirRelease: "$buildDir/intermediates/assets/release",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in debug mode
 *   resourcesDirDebug: "$buildDir/intermediates/res/merged/debug",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in release mode
 *   resourcesDirRelease: "$buildDir/intermediates/res/merged/release",
 *
 *   // by default the gradle tasks are skipped if none of the JS files or assets change; this means
 *   // that we don't look at files in android/ or ios/ to determine whether the tasks are up to
 *   // date; if you have any other folders that you want to ignore for performance reasons (gradle
 *   // indexes the entire tree), add them here. Alternatively, if you have JS files in android/
 *   // for example, you might want to remove it from here.
 *   inputExcludes: ["android/**", "ios/**"],
 *
 *   // override which node gets called and with what additional arguments
 *   nodeExecutableAndArgs: ["node"],
 *
 *   // supply additional arguments to the packager
 *   extraPackagerArgs: []
 * ]
 */

project.ext.react = [
    entryFile: "index.js"
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion
    buildToolsVersion rootProject.ext.buildToolsVersion

    defaultConfig {
        applicationId "com.app"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        missingDimensionStrategy "RNN.reactNativeVersion", "reactNative57_5" // See note below! // for react-native-navigation
        versionCode 1
        versionName "1.0"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
    }
    compileOptions { // for react-native-navigation
        sourceCompatibility JavaVersion.VERSION_1_8 // for react-native-navigation
        targetCompatibility JavaVersion.VERSION_1_8 // for react-native-navigation
    } // for react-native-navigation
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
}

dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation project(':react-native-navigation') // for react-native-navigation
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.compile
    into 'libs'
}
```
   
   
   
MainActivity.java     
```java
package com.app;

// import com.facebook.react.ReactActivity;


// public class MainActivity extends ReactActivity {

//     /**
//      * Returns the name of the main component registered from JavaScript.
//      * This is used to schedule rendering of the component.
//      */
//     @Override
//     protected String getMainComponentName() {
//         return "app";
//     }
// }


import com.reactnativenavigation.NavigationActivity; // for react-native-navigation
public class MainActivity extends NavigationActivity { // for react-native-navigation
} // for react-native-navigation
```   
      
     
   
MainApplication.java   
```java
package com.app;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;


import com.reactnativenavigation.NavigationApplication; // for react-nativa-navigation
import com.reactnativenavigation.react.NavigationReactNativeHost; // for react-nativa-navigation
import com.reactnativenavigation.react.ReactGateway; // for react-nativa-navigation


import java.util.Arrays;
import java.util.List;

// public class MainApplication extends Application implements ReactApplication {

//   private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
//     @Override
//     public boolean getUseDeveloperSupport() {
//       return BuildConfig.DEBUG;
//     }

//     @Override
//     protected List<ReactPackage> getPackages() {
//       return Arrays.<ReactPackage>asList(
//           new MainReactPackage()
//       );
//     }

//     @Override
//     protected String getJSMainModuleName() {
//       return "index";
//     }
//   };

//   @Override
//   public ReactNativeHost getReactNativeHost() {
//     return mReactNativeHost;
//   }

//   @Override
//   public void onCreate() {
//     super.onCreate();
//     SoLoader.init(this, /* native exopackage */ false);
//   }
// }

public class MainApplication extends NavigationApplication { // for-react-native-navigation
    
    @Override
    protected ReactGateway createReactGateway() {
        ReactNativeHost host = new NavigationReactNativeHost(this, isDebug(), createAdditionalReactPackages()) {
            @Override
            protected String getJSMainModuleName() {
                return "index";
            }
        };
        return new ReactGateway(this, isDebug(), host);
    }

    @Override
    public boolean isDebug() {
        return BuildConfig.DEBUG;
    }

    protected List<ReactPackage> getPackages() {
        // Add additional packages you require here
        // No need to add RnnPackage and MainReactPackage
        return Arrays.<ReactPackage>asList(
            // eg. new VectorIconsPackage()
        );
    }
  
    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
        return getPackages();
    }

} // for-react-native-navigation
``` 
   
   
   
gradle-wrapper.properties    
```
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
   
   
   
   
index.js   
```js
/** @format */

import { Navigation } from "react-native-navigation";
// import {AppRegistry} from 'react-native';
import App from '@src/App';
import {name as appName} from '@src/app.json';

// AppRegistry.registerComponent(appName, () => App);

Navigation.registerComponent(`navigation.playground.WelcomeScreen`, () => App);

Navigation.events().registerAppLaunchedListener(() => {
  Navigation.setRoot({
    root: {
      component: {
        name: "navigation.playground.WelcomeScreen"
      }
    }
  });
});
```
   
   
package.json
```
{
  "name": "app",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest",
    "android": "cd ./android && gradlew app:assembleDebug && gradlew installDebug"
  },
  "dependencies": {
    "react": "16.6.3",
    "react-native": "0.57.8",
    "react-native-navigation": "^2.4.0"
  },
  "devDependencies": {
    "@types/react": "^16.7.18",
    "@types/react-native": "^0.57.24",
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.51.1",
    "react-native-typescript-transformer": "^1.2.11",
    "react-test-renderer": "16.6.3",
    "ts-jest": "^23.10.5",
    "tslint": "^5.12.0",
    "typescript": "^3.2.2"
  },
  "jest": {
    "preset": "react-native"
  }
}

```   
### Environment
* React Native Navigation version: 2.4.0
* React Native version: 0.57.8
* Platform(s) (iOS, Android, or both?): android
* Device info (Simulator/Device? OS version? Debug/Release?): android 8.0.0 / Windows 10

--- a/lib/android/app/build.gradle
+++ b/lib/android/app/build.gradle
@@ -1,13 +1,21 @@
 apply plugin: 'com.android.library'
 apply from: '../prepare-robolectric.gradle'
 
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
+def DEFAULT_COMPILE_SDK_VERSION = 26
+def DEFAULT_MIN_SDK_VERSION = 19
+def DEFAULT_SUPPORT_LIB_VERSION = '26.1.0'
+def DEFAULT_TARGET_SDK_VERSION = 25
+
 android {
-    compileSdkVersion 26
-    buildToolsVersion '27.0.3'
+    compileSdkVersion safeExtGet('compileSdkVersion', DEFAULT_COMPILE_SDK_VERSION)
 
     defaultConfig {
-        minSdkVersion 19
-        targetSdkVersion 25
+        minSdkVersion safeExtGet('minSdkVersion', DEFAULT_MIN_SDK_VERSION)
+        targetSdkVersion safeExtGet('targetSdkVersion', DEFAULT_TARGET_SDK_VERSION)
         versionCode 1
         versionName "1.0"
     }
@@ -81,11 +89,13 @@ allprojects { p ->
     }
 }
 
+def supportLibVersion = safeExtGet('supportLibVersion', DEFAULT_SUPPORT_LIB_VERSION)
+
 dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
-    implementation 'com.android.support:design:26.1.0'
-    implementation 'com.android.support:appcompat-v7:26.1.0'
-    implementation 'com.android.support:support-v4:26.1.0'
+    implementation "com.android.support:design:${supportLibVersion}"
+    implementation "com.android.support:appcompat-v7:${supportLibVersion}"
+    implementation "com.android.support:support-v4:${supportLibVersion}"
 
     implementation 'com.github.wix-playground:ahbottomnavigation:2.4.9'
     implementation 'com.github.wix-playground:reflow-animator:1.0.4'

